doc/guides/prog_guide/cnet.rst

```diff
@@ -0,0 +1,400 @@
+
+- **eth_tx-N** is the node to send packets out a lport via the *Pktdev* APIs.
+
+  - **Note**: More then one output node may be present one per lport.
+
+- **arp_request** is the node to send request the kernel stack send and process ARP packets needing to be sent by CNET.
+
+  - ARP packet processing is handled by the Linux kernel and not by CNET. When the CNET stack needs a destination MAC address and it is not known a ARP request is sent by the kernel.
+
+- **chnl_callback** is the node to callback into the application to allow for more packet data processing to continue.
+
+  - When a channel is created a callback routine is given to allow the CNET to callback into the application to help complete the connection or data processing.
+
+Anatomy of CNET processing
+--------------------------
+
+The anatomy of the CNET stack is complex, but most of the processing and configuration is defined by
+the :ref:`Graph Library <Graph_Library>`, but a number of internal structures and designs should be understood.
+One of the design details is the CNET stack uses a couple thread local variables *this_cnet* and *this_stk*
+to help avoid passing these values in the APIs.
+
+One structure per application process using the CNET stack is the *struct cnet* or *stk_t* structure, which
+contains information and data accessed by the stack instances. The *struct stk_entry* structure contains data
+and information about each instance of the stack, plus it uses information from the *cnet* structure. A list of
+stack information structures is held by the *cnet* structure.
+
+The *cnet* structure contains global application information about the multiple instances of each stack.
+Each stack instance is independent of each other except for the needing information from the *cnet* structure.
+The *stk_t* structure instances that can contain different graph nodes or configuration to process
+different types of stack configurations and the *cnet-graph* example creates duplicate graphs one per
+stack instance.
+
+.. note:: For the internals of the *cnet* and *stk_t* structures refer to the *Doxygen* created API
+  documentation. The *channel* APIs are also defined in the documentation.
+
+The application uses the Channel APIs to create connections or setup listening connections similar
+to the socket API. Using APIs like *channel()*, *chnl_recv()*, *chnl_send()*, *chnl_bind()*, *chnl_listen()*,
+*chnl_connect()*, *chnl_open()* and *chnl_accept()* are a few of the APIs to use for creating connections. As
+can be seen the APIs look similar to standard *Socket* APIs.
+
+.. _figure_cnet_stack_view:
+
+.. figure:: img/cnet_stack_view.*
+
+   CNET Stack High Level View
+
+CNET Structure
+^^^^^^^^^^^^^^
+
+The CNET structure :ref:`CNET Structure <figure_cnet_structure>` contains number of fields to describe
+the information to control the CNET stack. Some of the details are described later and this structure
+is create once for all stack instances.
```

Comment:
"This structure is created once for all stack instances", remove the "some of the details are desscribed later.."